pkg/api/types.go

```diff
@@ -646,15 +648,12 @@ type PodCondition struct {
 	Status ConditionStatus  `json:"status"`
 }
 
-// PodInfo contains one entry for every container with available info.
-type PodInfo map[string]ContainerStatus
-
 // PodContainerInfo is a wrapper for PodInfo that can be encode/decoded
 // DEPRECATED: Replaced with PodStatusResult
 type PodContainerInfo struct {
 	TypeMeta      `json:",inline"`
 	ObjectMeta    `json:"metadata,omitempty"`
-	ContainerInfo PodInfo `json:"containerInfo"`
+	ContainerInfo []ContainerStatus `json:"containerInfo"`
```

Comment:
In terms of complexity I can't see any actual difficulty when using list, however all code that touches that list is modified in this PR, so if you feel it makes code more complex I can change it.

But I personally think that the code is more complicated when you have two different representations (in this case internal and external ones) of them same object, rather than using just not the best data structure, so I prefer to leave the code as it is now.